src/Convert/ExprUtils.hs

```diff
@@ -81,41 +81,41 @@ flattenConcat (e : es) =
 flattenConcat [] = []
 
 
-simplifyBinOp :: BinOp -> Expr -> Expr -> Expr
-
-simplifyBinOp Sub e (Dec 0) = e
-simplifyBinOp Sub (Dec 0) e = UniOp UniSub e
-simplifyBinOp Mul (Dec 0) _ = toDec 0
-simplifyBinOp Mul _ (Dec 0) = toDec 0
-simplifyBinOp Mod _ (Dec 1) = toDec 0
-
-simplifyBinOp Add e1 (UniOp UniSub e2) = BinOp Sub e1 e2
-simplifyBinOp Add (UniOp UniSub e1) e2 = BinOp Sub e2 e1
-simplifyBinOp Sub e1 (UniOp UniSub e2) = BinOp Add e1 e2
-simplifyBinOp Sub (UniOp UniSub e1) e2 = UniOp UniSub $ BinOp Add e1 e2
-
-simplifyBinOp Add (BinOp Add e n1@Number{}) n2@Number{} =
-    BinOp Add e (BinOp Add n1 n2)
-simplifyBinOp Sub n1@Number{} (BinOp Sub n2@Number{} e) =
-    BinOp Add (BinOp Sub n1 n2) e
-simplifyBinOp Sub n1@Number{} (BinOp Sub e n2@Number{}) =
-    BinOp Sub (BinOp Add n1 n2) e
-simplifyBinOp Sub (BinOp Add e n1@Number{}) n2@Number{} =
-    BinOp Add e (BinOp Sub n1 n2)
-simplifyBinOp Add n1@Number{} (BinOp Add n2@Number{} e) =
-    BinOp Add (BinOp Add n1 n2) e
-simplifyBinOp Add n1@Number{} (BinOp Sub e n2@Number{}) =
-    BinOp Add e (BinOp Sub n1 n2)
-simplifyBinOp Sub (BinOp Sub e n1@Number{}) n2@Number{} =
-    BinOp Sub e (BinOp Add n1 n2)
-simplifyBinOp Add (BinOp Sub e n1@Number{}) n2@Number{} =
-    BinOp Sub e (BinOp Sub n1 n2)
-simplifyBinOp Add (BinOp Sub n1@Number{} e) n2@Number{} =
-    BinOp Sub (BinOp Add n1 n2) e
-simplifyBinOp Ge (BinOp Sub e (Dec 1)) (Dec 0) = BinOp Ge e (toDec 1)
+simplifyBinOp :: BinOp -> [Attr] -> Expr -> Expr -> Expr
```

Comment:
I'm not sure if any of the `simplifyBinOp` cases are important for attributed expressions. Can you think of any exceptions?

Reply:
I'm going to look into that after reverting the changes to remove the attributes merging and moving.